src/TransactionsBundle/Service/MatchService.php

```diff
@@ -28,37 +28,51 @@ public function __construct(
         $this->entityManager = $entityManager;
     }
 
-    public function match($matches, $transaction, $category)
+    public function match($matches, $openTransaction, $category)
     {
-        $matchDescription = $this->cleanUp($transaction->getDescription());
-        $type = $this->categoryRepository->findById($category);
-
-        if ($transaction->getCategories() != null) {
+        if ($openTransaction->getCategories() != null) {
             return;
         }
 
+        $results = array();
+        $transactions = array();
+
+        $category = $this->categoryRepository->findOneById($category);
+        $matchDescription = $this->cleanUp($openTransaction->getDescription());
+
         foreach ($matches as $match) {
-            if ($type[0]->getAccountId() != $match->getAccountId()) {
+            /*
+             * If matches come from UI
+             * TODO SHOULD CLEAN THIS
+             */
+            if (gettype($match) === 'array') {
+                $match = $this
+                    ->entityManager
+                    ->getRepository('TransactionsBundle:Transactions')
+                    ->findOneById($match['id']);
+            }
+
+            if ($category->getAccountId() != $match->getAccountId()) {
```

Comment:
In your `$category = $this->categoryRepository->findOneById($category);` maybe you can add the accountid already in the select statement - this way you skip this check